### ISSUE TITLE
16 frames test

### ISSUE THREAD

**Issue by yangzhifeio** [CIntellifusion/GeometryForcing#9]
Hi,

I ran the following command:

bash scripts/eval_geometry_forcing_rotation.sh

The evaluation finished successfully, but the results are quite different from the 16-frame results reported in Table 1 of the paper.
Could you please clarify how the 16-frame experiment in Table 1 was obtained?
Specifically:

Does eval_geometry_forcing_rotation.sh correspond to that experiment?

Were any parameters (e.g., frame window size, keyframe settings, or dataset split) different from the default script?

Thanks in advance!

  | Framers | FID⬇️ | FVD⬇️ | LPIPS⬇️ | SSIM⬆️ | PSNR⬆️
GeometryForcing(paper model) | 16 | 36.0625 | 921 | 0.66 | 0.26 | 9.76

**Comment by CIntellifusion** [comment]
I have tested on the released checkpoint and the number is correct using `eval_geometry_forcing_rotation.sh `. Could u please share some example generated in this evaluation. I guess the FVD about 921  and PNSR about 9.76 means the checkpoint is not coreectly loaded and the video maybe completely in a mess. 

1. please check if the checkpint is correctly download and loaded into the model. If you are not sure, please copy and paste the output in terminal to here. 
2. check the video genreated are reasonable, which means it should contain clear room scene. 
3. Maybe u could try to inference with default dfot checkpoint to see the result using eval_geometry_forcing_rotation.sh

**Comment by wnqwang243** [comment]
I also ran with the provided checkpoint and code (scripts/eval_geometry_forcing_ratation.sh) for the 16 frames generation on the full real-estate-10k dataset, but the evaluation results don't match (worse than) the ones reported in the paper (Table 1). I followed the original code to load the checkpoint, and the generated videos look ok. Could you please explain this and perhaps are there different parameters you used? Thanks. The evaluation results I got are:    

 prediction/fid              35.28125
     prediction/fvd                921.5
    prediction/lpips         0.661293625831604
     prediction/psnr         9.81535530090332
     prediction/ssim         0.268154501914978

**Comment by CIntellifusion** [comment]
```
echo "Running 16 frame evaluation for DFoT on RealEstate10K dataset"
output_dir="evaluations"
eval_result_dir="$output_dir/geometry_forcing_short"
checkpoint_path="checkpoints/geometry_forcing_state_dict.ckpt"

algorithm="dfot_geometry_forcing"

echo "Result directory: $result_dir"
echo "Checkpoint path: $checkpoint_path" 

python -m main +name=single_image_to_long dataset=realestate10k \
        algorithm=$algorithm experiment=video_generation \
        @diffusion/continuous \
        load=$checkpoint_path \
        'experiment.tasks=[validation]' experiment.validation.data.shuffle=False experiment.test.data.shuffle=False \
        dataset.context_length=1 dataset.frame_skip=1 dataset.n_frames=16 \
        algorithm.tasks.interpolation.max_batch_size=4 experiment.validation.batch_size=1 \
        algorithm.tasks.prediction.history_guidance.name=stabilized_vanilla \
        +algorithm.tasks.prediction.history_guidance.guidance_scale=4.0 \
        +algorithm.tasks.prediction.history_guidance.stabilization_level=0.02  \
        algorithm.tasks.interpolation.history_guidance.name=vanilla \
        +algorithm.tasks.interpolation.history_guidance.guidance_scale=1.5 \
        'algorithm.logging.metrics=[fvd,fid,psnr,lpips,ssim]' \
        hydra.run.dir=$result_dir
```

To get 16 frame evaluation result, you could directly run this command using dataset realestate10k

The command you use is for generating videos that rotate for 360 degree, where is no ground truth video thus the metric is meaningless. 

To reproduce the 256 frame result, you could run `scripts/eval_geometry_forcing.sh`
To reproduce the 16 frame result, you could run `scripts/run_geometry_forcing.sh`

**Comment by Hippogriff** [comment]
There is no file by the name `scripts/run_geometry_forcing.sh`. Perhaps that file has not been uploaded to the repo yet.
To expedite, can you please just paste the script here.

**Comment by CIntellifusion** [comment]
Sorry I mean, 'scripts/train_geometry_forcing.sh` , the evaluation is performed after each epoch.

**Comment by wnqwang243** [comment]
I ran the`scripts/train_geometry_forcing.sh` on the full RE10K, but in the evaluation result, the FVD I got (611.5) seems higher than the one in the paper (243):      

prediction/fid               73.8125
     prediction/fvd                611.5
    prediction/lpips        0.4915119707584381
     prediction/psnr         11.96070671081543
     prediction/ssim        0.3903127908706665

**Comment by CIntellifusion** [comment]
the fvd 611 is much worse than the initial checkpoint of DFoT. The reasonable fvd value range should be around 250 or less. I suggest you to check 
1. if all the checkpoint is loaded correctly including vggt and dfot checkpoint. 
2. the generated video quality. 611 often means the video crashed. 
3. The logged training loss should be around 0.0x . 
4. The training environment, including cuda devices, cuda version etc.

**Comment by CIntellifusion** [comment]
> ```
> echo "Running 16 frame evaluation for DFoT on RealEstate10K dataset"
> output_dir="evaluations"
> eval_result_dir="$output_dir/geometry_forcing_short"
> checkpoint_path="checkpoints/geometry_forcing_state_dict.ckpt"
> 
> algorithm="dfot_geometry_forcing"
> 
> echo "Result directory: $result_dir"
> echo "Checkpoint path: $checkpoint_path" 
> 
> python -m main +name=single_image_to_long dataset=realestate10k \
>         algorithm=$algorithm experiment=video_generation \
>         @diffusion/continuous \
>         load=$checkpoint_path \
>         'experiment.tasks=[validation]' experiment.validation.data.shuffle=False experiment.test.data.shuffle=False \
>         dataset.context_length=1 dataset.frame_skip=1 dataset.n_frames=16 \
>         algorithm.tasks.interpolation.max_batch_size=4 experiment.validation.batch_size=1 \
>         algorithm.tasks.prediction.history_guidance.name=stabilized_vanilla \
>         +algorithm.tasks.prediction.history_guidance.guidance_scale=4.0 \
>         +algorithm.tasks.prediction.history_guidance.stabilization_level=0.02  \
>         algorithm.tasks.interpolation.history_guidance.name=vanilla \
>         +algorithm.tasks.interpolation.history_guidance.guidance_scale=1.5 \
>         'algorithm.logging.metrics=[fvd,fid,psnr,lpips,ssim]' \
>         hydra.run.dir=$result_dir
> ```
> 
> To get 16 frame evaluation result, you could directly run this command using dataset realestate10k
> 
> The command you use is for generating videos that rotate for 360 degree, where is no ground truth video thus the metric is meaningless.
> 
> To reproduce the 256 frame result, you could run `scripts/eval_geometry_forcing.sh` To reproduce the 16 frame result, you could run `scripts/run_geometry_forcing.sh`
This command could preduce in this result. Which is little lower than that in paper because the video sample interval may influence the generation result. Here, in this setting the video sample interval is 1 (see `frame_skip=1` in `configurations/dataset/base_video.yaml`) , where in training the `frame_skip=10`
<img width="525" height="152" alt="Image" src="https://github.com/user-attachments/assets/a834c9dd-c5b0-426e-b0e6-6b5555d64104" />

**Comment by Riga2** [comment]
Hi, thank you for your wonderful work!
I ran the script "eval_geometry_forcing_ratation.sh" to test the 360 rotation results.
But it seems the results are a little bit strange, and I report two result video below (I only changed the result video fps=1 for better visualization. Left is the generated video I think). 
Do you think the results are normal?

https://github.com/user-attachments/assets/1a53f523-a96a-453d-b90b-39b70721dd24

https://github.com/user-attachments/assets/2994b3c1-0e6e-4acb-8122-090f058e3774

**Comment by Hippogriff** [comment]
Running on 16 frames give but DfoT checkpoint (DFoT_16f_state_dict.ckpt), which is substantially higher than the numbers reported in the paper:

<img width="749" height="218" alt="Image" src="https://github.com/user-attachments/assets/6b3853f4-2405-4882-9495-fcbc1fd440d2" />
